### ISSUE TITLE
Support condition

### ISSUE THREAD

**Issue by smosek** [serverless-operations/serverless-step-functions#505]
# This is a Feature Proposal

## Description

For feature proposals:
* CloudFormation has a condition feature that can be used on resources to conditionally deploy them.
* Serverless supports this for lambda functions by using `condition: SomeCondition`
* It would be great for step functions to support the same syntax, with the result being a condition on the resource

**Comment by jiwoorico** [comment]
good point, I think we also need this feaure, as we have some lambda/step machines are only deployed into a specific region,  we could use the condition function on lambda, but for step machines, seems we don't have a good way to do this.
